File: Project 1/server_client_trial1/src/Message.java
public class Message{
    String message;
    String sender;
    String receiver;
    String subject;
    String date;

    Integer message_id;
    public Message(String message,String sender,String receiver,String subject,String date){
        this.message = message;
        this.sender = sender;
        this.receiver = receiver;
        this.subject = subject;
        this.date = date;
    }

    public Message(Integer message_id,String message,String sender,String receiver,String subject,String date){
        this.message_id = message_id;
        this.message = message;
        this.sender = sender;
        this.receiver = receiver;
        this.subject = subject;
        this.date = date;
    }

    public Message(String message,String receiver,String subject,String date){
        this.message = message;
        this.receiver = receiver;
        this.subject = subject;
        this.date = date;
    }

    public void setSender(String sender){
        this.sender = sender;
    }


}

File: Project 3/mean-stack-crud-app/backend/models/Message.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

//pagination//
var mongoosePaginate = require('mongoose-paginate');




// Define collection and schema
let MessageSchema = new Schema({
    sender: {
      type: String
    },
    receiver: {
      type: String
    },
    subject: {
      type: String
    },
    date: {
      type: Date
    },
    messageContent: {
      type: String
    }
  }, {
   collection: 'messages'
})

MessageSchema.plugin(mongoosePaginate);

var Model = mongoose.model('Model',  MessageSchema); // Model.paginate()


const Message = module.exports = mongoose.model('Message', MessageSchema);


module.exports.getMessageById = function(id,callback){
  Message.findById(id,callback);
}

module.exports.getMessageBySender = function(sender,callback){
  const query = {sender: sender};
  Sender.findOne(query,callback);
}

module.exports.addMessage = function(newMessage,callback){
  newMessage.save(callback);
}


